### PR TITLE
Doc restructure

### LINKED ISSUE #1951: Documentation: Restructure the help
Ever moving forward to a more user-friendly structure.

- Download
    - Versions
        - [Download + Release Notes] (for each version)
    - Notes for Packagers
- Quickstart
- About
    - Features
    - Concepts
    - Cyrus Board
    - Cyrus History
- Setup
    - Building
    - Installing
    - Upgrading
    - Configuration
- Operations
    - Man Pages
    - Admin Reference
    - SOP
    - Other tasks (maintenance, recovery)
- Developers
    - Developer Environment
    - Process
    - Debugging
    - Documentation contributions
- Community/Support
    - Mailing lists
    - IRC
    - Logging a bug

### PR DESCRIPTION
**Pull request by nicolan**
Fixes #1951 

* Cleaner top level help structure
* Splitting apart how to get, compile and install Cyrus into more manageable sections.

### CODE REVIEW

**Comment by postilion** [comment]
@nicolan I've grabbed your code and built new HTML & man pages from it.  Cruising through the HTML, however, I come across immediate problems.  Here's what I see right now:

- From Top level (i.e. index.html) menu on left reads, from top-down: Download | Quickstart | Overview | Setup | Operations | Developers | Support/Community

- Upon selecting Quickstart Guide, however, the menu changes to this: Quickstart | Download | Concepts | Setup | Operations | Developers | Support/Community  In other words, Quickstart moves to the top, and Overview changes to Concepts.

- Clicking on Concepts (from Quickstart) or Overview (from Concepts) leads to imap/concepts/overview_and_concepts.html, which is formatted all wrong (loses sidebar, for example).

- From index.html, the Overview menu item has Features and Concepts as sub-items.  From other screens, those where Overview is replaced in the left-hand menu with Concepts, however, Concepts has sub-items Features, Overview and Configuration Guide.

- Clicking on Configuration Guide changes the left-hand menu to show us under Setup, and Concepts changes back to Overview.

In short, there's some broken plumbing here.  In whole, I think you've captured what you and I discussed when I was there, that last day.  But with the menu plumbing broken, it's a bit hard to get around.

Honestly, one reason I hadn't jumped in to help with this restructuring is that I don't really have a firm grasp of how all of this sidebar and menu stuff is handled by Sphinx.  Sorry :-(

Cheers,
-nic

**Comment by nicolan** [comment]
Hey @postilion!

Thanks for giving it a look. Can I get you to try again with a make clean first?

The sidebar needs to be rebuilt for *each page*. It's not a single html sidebar that is included/iframed, it is generated for each page for the toc. So if a given page isn't rebuilt, it will retain its last set of knowledge of the toc, which can be out of date. A make clean should fix this.

**Comment by nicolan** [comment]
I can't reproduce @postilion's errors and neither can @elliefm. Sending to master and 3.0

**Comment by postilion** [comment]
Works fine for me now.  Go figure...
